dm/syncer/syncer.go

```diff
@@ -2196,7 +2211,7 @@ func (s *Syncer) Run(ctx context.Context) (err error) {
 		}
 
 		// set exitSafeModeTS when meet first binlog
-		if s.firstMeetBinlogTS == nil && s.cliArgs != nil && s.cliArgs.SafeModeDuration != "" {
+		if s.firstMeetBinlogTS == nil && s.cliArgs != nil && s.cliArgs.SafeModeDuration != "" && int64(e.Header.Timestamp) != 0 {
```

Comment:
we should skip formatDescription event too who's timestamp may much smaller than the timestamp of first trancaction event, it's sent on connect every time.